
# Workshop 3: Comparative genomics

## Friday, May 5 (10am-1pm)

**Instructor:** Yocelyn Gutiérrez-Guerrero (ygutierrezgro@berkeley.edu)

This workshop will use whole genome information of different species at different phylogenetic distances to explore how the genomes have evolved. Comparative genomics is a field that will analyze the synteny of genomes, reconstruct phylogenetic relationships using hundreds or thousands of genome regions, gene content, the evolution of gene families, and selection pressures across the species.

### Topics

**1. Which questions we can address using a comparative genomic approach (slides [here](XXX))**
* ***Activities and questions [here](XXX)***

**2. Evaluate the quality of a genome available in databases (slides [here](XXX))**
* ***Activities and questions [here](XXX)***

**3. Phylogenomics (slides [here](XXX))**
* Generating databases of orthologous genes
* Alignment, trimming and filtering sequences
* Evolutionary best fit model
* Tree construction
* ***Activities and questions [here](XXX)***

**4. Selection test comparing species (slides [here](XXX))**
* *dN/dS* selection test (Codeml and Hyphy)
* Identify single copy genes
* Null and Alternative models
* Likelihood test and sites under selection
* Functional annotation
* Gene Ontology enrichment
* ***Activities and questions [here](XXX)***

**5. Gene family evolution review (slides [here](XXX))**
* ***Activities and questions [here](XXX)***

**6. Introgression analysis using D-statistics (ABBA-BABA) (slides [here](XXX))**
* ***Activities and questions [here](XXX)***

### Software required
**Linux:** `Diamond`, `Orthofinder`, `OrthoMCL v.2`, `ProteinOrtho`, `Blast`, `Prottest 3`, `MAFT`, `MUSCLE`, `RAxML`, `Astral`, `CODEML`, `Hyphy`, `seaview`

**R packages:** `topGO`, `snpR`, `ape`

See course installation instructions [here](XXX).

### Example dataset to be used

We will be using XXX.

### Additional resources

Kapli, P., Yang, Z. & Telford, M.J. Phylogenetic tree building in the genomic age. Nat Rev Genet 21, 428–444 (2020). Link [here](https://doi.org/10.1038/s41576-020-0233-0).

Martin, S.H., Davey, J.W., Jiggins, C.D. Evaluating the use of ABBA-BABA statistics to locate introgressed loci. Mol Biol Evol. 32:244-57 (2015). Link [here](https://doi.org/10.1093/molbev/msu269).

Sabeti, PC., et al., Positive natural selection in the human lineage. Science 312 (2006). Link [here](http://doi.org/10.1126/science.1124309).
